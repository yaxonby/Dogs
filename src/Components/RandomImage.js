import React, { PureComponent } from 'react';
import Fab from '@material-ui/core/Fab';

export default class RandomImage extends PureComponent {
  constructor(props) {
    super(props);
    this.state = { randomFoto: null };
    this.loadData = this.loadData.bind(this);
  }

  loadData() {
    const self = this;
    fetch('https://dog.ceo/api/breeds/image/random')
      .then((response) => {
        if (response.status !== 200) {
          console.log(`Looks like there was a problem. Status Code: ${response.status}`);
          return;
        }
        response.json().then((data) => {
          self.setState({ randomFoto: data.message });
        });
      })
      .catch((err) => { console.log('Fetch Error :-S', err); });
  }

  componentDidMount() { this.loadData(); }

  render() {
    return (
      <div className='positionCenter'>
        <h3> View random image of dog</h3>
        <div>
          <img src={this.state.randomFoto}/>
          <Fab color='default' variant='round' onClick={this.loadData}> Next</Fab>
        </div>
      </div>
    );
  }
}
