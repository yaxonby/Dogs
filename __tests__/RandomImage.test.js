import React from 'react';
import RandomImage from '../src/Components/RandomImage';

it('should render a RandomImage', () => {
  const wrapper = shallow(
        <RandomImage />
  );
  expect(wrapper).toMatchSnapshot();
});
