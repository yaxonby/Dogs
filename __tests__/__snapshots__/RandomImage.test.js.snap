// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a RandomImage 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RandomImage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
           View random image of dog
        </h3>,
        <div>
          <img
            src={null}
          />
          <WithStyles(Fab)
            color="default"
            onClick={[Function]}
            variant="round"
          >
             Next
          </WithStyles(Fab)>
        </div>,
      ],
      "className": "positionCenter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " View random image of dog",
        },
        "ref": null,
        "rendered": " View random image of dog",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              src={null}
            />,
            <WithStyles(Fab)
              color="default"
              onClick={[Function]}
              variant="round"
            >
               Next
            </WithStyles(Fab)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": null,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": " Next",
              "color": "default",
              "onClick": [Function],
              "variant": "round",
            },
            "ref": null,
            "rendered": " Next",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
             View random image of dog
          </h3>,
          <div>
            <img
              src={null}
            />
            <WithStyles(Fab)
              color="default"
              onClick={[Function]}
              variant="round"
            >
               Next
            </WithStyles(Fab)>
          </div>,
        ],
        "className": "positionCenter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " View random image of dog",
          },
          "ref": null,
          "rendered": " View random image of dog",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                src={null}
              />,
              <WithStyles(Fab)
                color="default"
                onClick={[Function]}
                variant="round"
              >
                 Next
              </WithStyles(Fab)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "src": null,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": " Next",
                "color": "default",
                "onClick": [Function],
                "variant": "round",
              },
              "ref": null,
              "rendered": " Next",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
